### PR TITLE
Added 4 chapters.

### PR DESCRIPTION
**Pull request by snehasharma76**
Address, Vectors, Function, Signer

### CODE REVIEW

**Review by 0xNekr — CHANGES_REQUESTED** [review]
The aim of the collaboration is not simply to copy and paste the existing course into the Aptos Move Book. Even if the overall structure is the same in terms of learning order, our aim is to bring a different experience to users. 

You can take some of the information and copy it, like tables and examples, but you can't copy the lesson word for word.

We therefore need to rewrite and simplify what is sometimes crudely explained in the Move Book to make learning simple and fun, as was done for the previous steps. 

Otherwise, the exercises are pretty good. Just remember to space out the code. The inclusion of an island universe is appreciated.

Sources from which the 4 steps were copied : 

https://move-book.com/
https://aptos.dev/move/book/

File: sources/quest_1/chapter_1/step_8/step_8_target.move [hunk 1]
```diff
@@ -0,0 +1,13 @@
+module robinson::my_shore {
+
```
Comment (0xNekr) [inline sources/quest_1/chapter_1/step_8/step_8_target.move:2]: You should import lib on the top of the module 

```
module robinson::my_shore {

    use  std::signer;

    struct GlobalData has key {
        nb_tree: u8,
        has_river: bool,
        shore_location: address,
        nb_trees: vector<u8>,
    }

    fun shores(s: signer) {
        assert!(signer::address_of(&s) == @0x42, 0);
    }
}
```

**Review by 0xNekr — CHANGES_REQUESTED** [review]
Much better, I've added two comments, a suggestion and an improvement

File: sources/quest_1/chapter_1/step_6/step_6_target.move [hunk 2]
```diff
@@ -0,0 +1,10 @@
+module robinson::my_shore {
+
+    struct GlobalData has key {
+        nb_tree: u8,
+        has_river: bool,
+        shore_location: address,
+        nb_trees: vector<u8>,
```
[the remark this answers is not in the document]
Comment (snehasharma76) [reply sources/quest_1/chapter_1/step_6/step_6_target.move:7]: Noted!

File: sources/quest_1/chapter_1/step_8/step_8.md [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+
+**Using Signer in a Script:**
+
+    -   In a script (a set of instructions for the blockchain), you can use the `std::signer` module to perform operations on `signer` values.
+
+    -   The `signer::address_of(&signer)` function gives you the address wrapped by the signer, and `signer::borrow_address(&signer)` gives you a reference to the address.
+
+---
+
+Now it's our turn to add the `signer` as a parameter to our function `shore` and check if the signer of the function `shore` is equal to address of `@0x42` using `signer::address_of(&s) == @0x42`.
```
[the remark this answers is not in the document]
Comment (snehasharma76) [reply sources/quest_1/chapter_1/step_8/step_8.md:23]: Missed that, my bad!

File: sources/quest_1/chapter_1/step_8/step_8_target.move (hunk 1 above)
Comment (snehasharma76) [reply sources/quest_1/chapter_1/step_8/step_8_target.move:2]: Noted!

**Review by 0xNekr — APPROVED** [review]

**Pull request merged**